src/utils/apiClient.tsx

```
@@ -0,0 +1,347 @@
+import React, { useState, useEffect, useRef } from "react";
+
+interface ApiResponse<T = any> {
+  data?: T;
+  error?: string;
+  status: number;
+  headers?: Record<string, string>;
+}
+
+interface RequestConfig {
+  timeout?: number;
+  retries?: number;
+  cache?: boolean;
+  skipAuth?: boolean;
+}
+
+class ApiClient {
+  private baseUrl: string;
+  private defaultConfig: RequestConfig;
+  private cache: Map<string, { data: any; timestamp: number; ttl: number }>;
+
+  constructor(baseUrl: string = "/api") {
+    this.baseUrl = baseUrl;
+    this.defaultConfig = {
+      timeout: 10000,
+      retries: 3,
+      cache: true,
+      skipAuth: false
+    };
+    this.cache = new Map();
+  }
+
+  private async makeRequest<T>(
+    endpoint: string,
+    options: RequestInit = {},
+    config: RequestConfig = {}
+  ): Promise<ApiResponse<T>> {
+    const finalConfig = { ...this.defaultConfig, ...config };
+    const url = `${this.baseUrl}${endpoint}`;
+    const cacheKey = `${options.method || "GET"}:${url}:${JSON.stringify(
+      options.body
+    )}`;
+
+    // Check cache first
+    if (finalConfig.cache && options.method === "GET") {
+      const cached = this.cache.get(cacheKey);
+      if (cached && Date.now() - cached.timestamp < cached.ttl) {
+        return { data: cached.data, status: 200 };
+      }
+    }
+
+    const headers: Record<string, string> = {
+      "Content-Type": "application/json",
+      ...((options.headers as Record<string, string>) || {})
+    };
+
+    // Add auth header - but doesn't validate token expiry
+    if (!finalConfig.skipAuth) {
+      const token = localStorage.getItem("authToken");
+      if (token) {
+        headers["Authorization"] = `Bearer ${token}`;
+      }
```

Should we throw an error or redirect user to login page if token is missing.